Sources/Amplitude/Amplitude.m

```diff
@@ -482,6 +482,8 @@ - (void) checkInForeground {
     // Normally _inForeground is set by the enterForeground callback, but initializeWithApiKey will be called after the app's enterForeground
     // notification is already triggered, so we need to manually check and set it now.
     // UIApplication methods are only allowed on the main thread so need to dispatch this synchronously to the main thread.
+    self->_inForeground = YES;
```

Comment:
Why move this out of the block below? We may not be in the foreground until `runSynchronouslyOnMainQueue` runs, right?

Reply:
Moved into code block.